internal/dev_server/sdk/streaming.go

```diff
@@ -67,3 +75,21 @@ func OpenStream(w http.ResponseWriter, done <-chan struct{}, initialMessage Mess
 	}()
 	return updateChan, errChan
 }
+
+func SendMessage(
+	updateChan chan<- Message,
+	msgType MessageType,
+	data interface{},
+) error {
+	payload, err := json.Marshal(data)
+	if err != nil {
+		return err
+	}
+
+	updateChan <- Message{
+		Event: msgType,
+		Data:  payload,
+	}
+
+	return nil
+}
```

Comment:
Not really sure how / if to unit test this - I've used Go channels only sparingly.

On the other hand - emitting messages worked fine before this PR, and testing `json.Marshal` feels beyond superfluous

Reply:
Might be good to add to [the integration tests](https://github.com/launchdarkly/ldcli/blob/37dddb611caa73c95e05d915d64812f889d139be/internal/dev_server/sdk/go_sdk_test.go#L27) for syncing the project.

RE: unit testing, you _could_ do something like the following.

```go
...
func TestSendMessage(t *testing.T) {
    chan := make(chan Message, 1) // needs to be buffered or else you'll deadlock
    err := SendMessage(chan, PUT_TYPE, "cat")
    require.NoError(t, err)
    message := <-chan
    ...do assertions
}
```

I don't think it's super important to unit test this function: it doesn't really have any logic & it is already being exercised through the integration tests.

Reply:
I decided to implement the integration test for anyone reading this after the fact